lib/commands/actions.js

```diff
@@ -169,33 +172,68 @@ commands.fingerprint = async function (fingerprintId) {
   await this.adb.fingerprint(fingerprintId);
 };
 
-commands.getScreenshot = async function () {
-  let localFile = temp.path({prefix: 'appium', suffix: '.png'});
+async function storeScreenshotDataWithAdbShell (localFile) {
   let pngDir = this.opts.androidScreenshotPath || '/data/local/tmp/';
   const png = path.posix.resolve(pngDir, 'screenshot.png');
-  let cmd =  ['/system/bin/rm', `${png};`, '/system/bin/screencap', '-p', png];
+  let cmd = ['/system/bin/rm', `${png};`, '/system/bin/screencap', '-p', png];
   await this.adb.shell(cmd);
+  await this.adb.pull(png, localFile);
+}
+
+async function storeScreenshotDataWithAdbExecOut (localFile) {
+  let {stdout} = await exec(this.adb.executable.path, 'exec-out', '/system/bin/screencap -p');
+  if (_.isString(stdout) && stdout.toString('hex', 0, 4) === PNG_MAGIC) {
```

Comment:
I'd rather this condition be flipped, so the work on the function is outside the conditional altogether.